src/components/cards/CreatePool/ChooseWeights.vue

```diff
@@ -0,0 +1,213 @@
+<script setup lang="ts">
+import { configService } from '@/services/config/config.service';
+import TokenInput from '@/components/inputs/TokenInput/TokenInput.vue';
+import TokenWeightInput from '@/components/inputs/TokenInput/TokenWeightInput.vue';
+import useWeb3 from '@/services/web3/useWeb3';
+import { computed, onMounted, reactive, ref, nextTick } from 'vue';
+import useNumbers from '@/composables/useNumbers';
+import useBreakpoints from '@/composables/useBreakpoints';
+import anime from 'animejs';
+import { sum, sumBy } from 'lodash';
+import usePoolCreation, {
+  TokenWeight
+} from '@/composables/pools/usePoolCreation';
+
+const emit = defineEmits(['update:tokenWeights', 'nextStep']);
+
+const emptyTokenWeight: TokenWeight = {
+  tokenAddress: '',
+  weight: 0,
+  id: 0,
+  isLocked: false,
+  amount: 0
+};
+
+const { userNetworkConfig } = useWeb3();
+const { tokenWeights, updateTokenWeights, proceed } = usePoolCreation();
+const networkName = configService.network.name;
+
+const _tokenOutAmount = ref();
+const _tokenOutAddress = ref();
+
+const tokenWeightListWrapper = ref<HTMLElement>();
+const addTokenRowElement = ref<HTMLElement>();
+const totalsRowElement = ref<HTMLElement>();
+const tokenWeightItemElements = reactive<HTMLElement[]>([]);
+const wrapperHeight = ref(0);
+
+const { upToLargeBreakpoint } = useBreakpoints();
+const { fNum } = useNumbers();
+
+const tokenWeightItemHeight = computed(() =>
+  upToLargeBreakpoint.value ? 56 : 64
+);
+
+onMounted(async () => {
+  // retrieving height causes reflow, get the height of the wrapper once
+  // and manually uptick it when we add items to prevent double reflow during anim
+  wrapperHeight.value = tokenWeightListWrapper.value?.offsetHeight || 0;
+
+  // add in the first token list item
+  addTokenToPool();
+  addTokenToPool();
+
+  // wait for vue to reflect the changes of above
+  await nextTick();
+
+  distributeWeights();
+});
+
+const handleWeightChange = (weight: string, id: number) => {
+  const tokenWeight = tokenWeights.value[id];
+  tokenWeight.weight = Number(weight);
+
+  distributeWeights();
+};
+
+const handleAddressChange = (address: string, id: number) => {
+  const tokenWeight = tokenWeights.value[id];
+  tokenWeight.tokenAddress = address;
+};
+
+const handleLockedWeight = (isLocked: boolean, id: number) => {
+  const tokenWeight = tokenWeights.value[id];
+  tokenWeight.isLocked = isLocked;
+  distributeWeights();
+};
```

Comment:
Why use arrow functions over regular functions here? I find that using regular functions helps to distinguish them from computed properties etc.